apps/smart-contracts/token/test/MiniSales.test.ts

```diff
@@ -771,5 +771,112 @@ describe('=> MiniSales', () => {
+        )
+        expect(await ppo.balanceOf(miniSales.address)).to.eq(
+          miniSalesBalanceBefore.sub(saleTokenAmount)
+        )
+      })
+
+      it('reverts if user transfers purchased PPO to blocked recipient', async () => {
+        expect(await blocklist.isIncluded(blockedUser.address)).to.eq(true)
+
+        await expect(ppo.connect(allowedSourceUser).transfer(blockedUser.address, 1)).revertedWith(
+          'Recipient blocked'
+        )
+      })
+
+      it('allows user to transfers purchased PPO to not allowed destination', async () => {
+        expect(await destinationAllowlist.isIncluded(disallowedDestinationUser.address)).to.eq(
+          false
+        )
+        const disallowedDestinationUserBalanceBefore = await ppo.balanceOf(
+          disallowedDestinationUser.address
+        )
+        const allowedSourceUserBalanceBefore = await ppo.balanceOf(allowedSourceUser.address)
+
+        await ppo.connect(allowedSourceUser).transfer(disallowedDestinationUser.address, 1)
+
+        expect(await ppo.balanceOf(disallowedDestinationUser.address)).to.eq(
+          disallowedDestinationUserBalanceBefore.add(1)
+        )
+        expect(await ppo.balanceOf(allowedSourceUser.address)).to.eq(
+          allowedSourceUserBalanceBefore.sub(1)
+        )
+      })
+
+      it('allows user to transfer purchased PPO to allowed destination', async () => {
+        expect(await destinationAllowlist.isIncluded(allowedDestinationUser.address)).to.eq(true)
+        const allowedDestinationUserBalanceBefore = await ppo.balanceOf(
+          allowedDestinationUser.address
+        )
+        const allowedSourceUserBalanceBefore = await ppo.balanceOf(allowedSourceUser.address)
+
+        await ppo.connect(allowedSourceUser).transfer(allowedDestinationUser.address, 1)
+
+        expect(await ppo.balanceOf(allowedDestinationUser.address)).to.eq(
+          allowedDestinationUserBalanceBefore.add(1)
+        )
+        expect(await ppo.balanceOf(allowedSourceUser.address)).to.eq(
+          allowedSourceUserBalanceBefore.sub(1)
+        )
+      })
+
+      it('allows user to purchase again', async () => {
+        const allowedSourceUserBalanceAfterFirstPurchase = await ppo.balanceOf(
+          allowedSourceUser.address
+        )
+        const miniSalesBalanceAfterFirstPurchase = await ppo.balanceOf(miniSales.address)
+
+        await miniSales.connect(purchaser).purchase(allowedSourceUser.address, 1, testPrice)
+
+        expect(await ppo.balanceOf(allowedSourceUser.address)).to.eq(
+          allowedSourceUserBalanceAfterFirstPurchase.add(1)
+        )
+        expect(await ppo.balanceOf(miniSales.address)).to.eq(
+          miniSalesBalanceAfterFirstPurchase.sub(1)
+        )
+      })
+
+      describe('if permissioned sale', () => {
+        beforeEach(async () => {
+          await setupPermissionedSale()
+        })
+
+        it('reverts purchase if user is not allowed recipient', async () => {
```

Comment:
```suggestion
        it('reverts purchase if user is not allowed sale recipient', async () => {
```
Do this for the other one